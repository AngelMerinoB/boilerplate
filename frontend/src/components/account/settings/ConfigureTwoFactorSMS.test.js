import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { ConfigureTwoFactorSMS } from './ConfigureTwoFactorSMS';
Enzyme.configure({ adapter: new Adapter() });
const noop = () => {};

describe('<ConfigureTwoFactorSMS />', () => {
  test('should render <ConfigureTwoFactorSMS /> without errors', () => {
    const wrapper = Enzyme.shallow(
      <ConfigureTwoFactorSMS
        location={{ search: '' }}
        generateTwoFactorSMS={noop}
        configureTwoFactorSMS={noop}
        user={{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the confirm message', () => {
    const wrapper = Enzyme.shallow(
      <ConfigureTwoFactorSMS
        location={{ search: '?confirm' }}
        generateTwoFactorSMS={noop}
        configureTwoFactorSMS={noop}
        user={{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render the ok message', () => {
    const wrapper = Enzyme.shallow(
      <ConfigureTwoFactorSMS
        location={{ search: '?sent' }}
        generateTwoFactorSMS={noop}
        configureTwoFactorSMS={noop}
        user={{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
