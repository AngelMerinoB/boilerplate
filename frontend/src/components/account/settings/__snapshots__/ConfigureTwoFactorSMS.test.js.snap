// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfigureTwoFactorSMS /> should render <ConfigureTwoFactorSMS /> without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfigureTwoFactorSMS
    configureTwoFactorSMS={[Function]}
    generateTwoFactorSMS={[Function]}
    location={
        Object {
            "search": "",
          }
    }
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Form
        ErrorList={[Function]}
        liveValidate={false}
        noHtml5Validate={true}
        noValidate={false}
        onSubmit={[Function]}
        safeRenderCompletion={false}
        schema={
                Object {
                        "properties": Object {
                          "phoneNumber": Object {
                            "title": "What's your phone number",
                            "type": "string",
                          },
                        },
                        "required": Array [
                          "phoneNumber",
                        ],
                        "title": "Add SMS Authentication",
                        "type": "object",
                      }
        }
        uiSchema={
                Object {
                        "phoneNumber": Object {},
                      }
        }
>
        <fieldset>
                <button
                        className="btn btn-primary"
                        type="submit"
                >
                        Send authentication code
                </button>
        </fieldset>
</Form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ErrorList": [Function],
        "children": Array [
          undefined,
          <fieldset>
            <button
                        className="btn btn-primary"
                        type="submit"
            >
                        Send authentication code
            </button>
</fieldset>,
        ],
        "liveValidate": false,
        "noHtml5Validate": true,
        "noValidate": false,
        "onSubmit": [Function],
        "safeRenderCompletion": false,
        "schema": Object {
          "properties": Object {
            "phoneNumber": Object {
              "title": "What's your phone number",
              "type": "string",
            },
          },
          "required": Array [
            "phoneNumber",
          ],
          "title": "Add SMS Authentication",
          "type": "object",
        },
        "uiSchema": Object {
          "phoneNumber": Object {},
        },
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-primary"
              type="submit"
>
              Send authentication code
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Send authentication code",
              "className": "btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Send authentication code",
            "type": "button",
          },
          "type": "fieldset",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Form
          ErrorList={[Function]}
          liveValidate={false}
          noHtml5Validate={true}
          noValidate={false}
          onSubmit={[Function]}
          safeRenderCompletion={false}
          schema={
                    Object {
                              "properties": Object {
                                "phoneNumber": Object {
                                  "title": "What's your phone number",
                                  "type": "string",
                                },
                              },
                              "required": Array [
                                "phoneNumber",
                              ],
                              "title": "Add SMS Authentication",
                              "type": "object",
                            }
          }
          uiSchema={
                    Object {
                              "phoneNumber": Object {},
                            }
          }
>
          <fieldset>
                    <button
                              className="btn btn-primary"
                              type="submit"
                    >
                              Send authentication code
                    </button>
          </fieldset>
</Form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ErrorList": [Function],
          "children": Array [
            undefined,
            <fieldset>
              <button
                            className="btn btn-primary"
                            type="submit"
              >
                            Send authentication code
              </button>
</fieldset>,
          ],
          "liveValidate": false,
          "noHtml5Validate": true,
          "noValidate": false,
          "onSubmit": [Function],
          "safeRenderCompletion": false,
          "schema": Object {
            "properties": Object {
              "phoneNumber": Object {
                "title": "What's your phone number",
                "type": "string",
              },
            },
            "required": Array [
              "phoneNumber",
            ],
            "title": "Add SMS Authentication",
            "type": "object",
          },
          "uiSchema": Object {
            "phoneNumber": Object {},
          },
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-primary"
                type="submit"
>
                Send authentication code
</button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Send authentication code",
                "className": "btn btn-primary",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Send authentication code",
              "type": "button",
            },
            "type": "fieldset",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<ConfigureTwoFactorSMS /> should render the confirm message 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfigureTwoFactorSMS
    configureTwoFactorSMS={[Function]}
    generateTwoFactorSMS={[Function]}
    location={
        Object {
            "search": "?confirm",
          }
    }
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Form
        ErrorList={[Function]}
        liveValidate={false}
        noHtml5Validate={true}
        noValidate={false}
        onSubmit={[Function]}
        safeRenderCompletion={false}
        schema={
                Object {
                        "properties": Object {
                          "token": Object {
                            "title": "Please enter the 6-digit authentication code we just sent you",
                            "type": "string",
                          },
                        },
                        "required": Array [
                          "token",
                        ],
                        "title": "Add SMS Authentication",
                        "type": "object",
                      }
        }
        uiSchema={
                Object {
                        "token": Object {},
                      }
        }
>
        <fieldset>
                <button
                        className="btn btn-primary"
                        type="submit"
                >
                        Verify code
                </button>
        </fieldset>
</Form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ErrorList": [Function],
        "children": Array [
          undefined,
          <fieldset>
            <button
                        className="btn btn-primary"
                        type="submit"
            >
                        Verify code
            </button>
</fieldset>,
        ],
        "liveValidate": false,
        "noHtml5Validate": true,
        "noValidate": false,
        "onSubmit": [Function],
        "safeRenderCompletion": false,
        "schema": Object {
          "properties": Object {
            "token": Object {
              "title": "Please enter the 6-digit authentication code we just sent you",
              "type": "string",
            },
          },
          "required": Array [
            "token",
          ],
          "title": "Add SMS Authentication",
          "type": "object",
        },
        "uiSchema": Object {
          "token": Object {},
        },
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-primary"
              type="submit"
>
              Verify code
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Verify code",
              "className": "btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Verify code",
            "type": "button",
          },
          "type": "fieldset",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Form
          ErrorList={[Function]}
          liveValidate={false}
          noHtml5Validate={true}
          noValidate={false}
          onSubmit={[Function]}
          safeRenderCompletion={false}
          schema={
                    Object {
                              "properties": Object {
                                "token": Object {
                                  "title": "Please enter the 6-digit authentication code we just sent you",
                                  "type": "string",
                                },
                              },
                              "required": Array [
                                "token",
                              ],
                              "title": "Add SMS Authentication",
                              "type": "object",
                            }
          }
          uiSchema={
                    Object {
                              "token": Object {},
                            }
          }
>
          <fieldset>
                    <button
                              className="btn btn-primary"
                              type="submit"
                    >
                              Verify code
                    </button>
          </fieldset>
</Form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ErrorList": [Function],
          "children": Array [
            undefined,
            <fieldset>
              <button
                            className="btn btn-primary"
                            type="submit"
              >
                            Verify code
              </button>
</fieldset>,
          ],
          "liveValidate": false,
          "noHtml5Validate": true,
          "noValidate": false,
          "onSubmit": [Function],
          "safeRenderCompletion": false,
          "schema": Object {
            "properties": Object {
              "token": Object {
                "title": "Please enter the 6-digit authentication code we just sent you",
                "type": "string",
              },
            },
            "required": Array [
              "token",
            ],
            "title": "Add SMS Authentication",
            "type": "object",
          },
          "uiSchema": Object {
            "token": Object {},
          },
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-primary"
                type="submit"
>
                Verify code
</button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Verify code",
                "className": "btn btn-primary",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Verify code",
              "type": "button",
            },
            "type": "fieldset",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<ConfigureTwoFactorSMS /> should render the ok message 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfigureTwoFactorSMS
    configureTwoFactorSMS={[Function]}
    generateTwoFactorSMS={[Function]}
    location={
        Object {
            "search": "?sent",
          }
    }
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Form
        ErrorList={[Function]}
        liveValidate={false}
        noHtml5Validate={true}
        noValidate={false}
        onSubmit={[Function]}
        safeRenderCompletion={false}
        schema={
                Object {
                        "properties": Object {
                          "phoneNumber": Object {
                            "title": "What's your phone number",
                            "type": "string",
                          },
                        },
                        "required": Array [
                          "phoneNumber",
                        ],
                        "title": "Add SMS Authentication",
                        "type": "object",
                      }
        }
        uiSchema={
                Object {
                        "phoneNumber": Object {},
                      }
        }
>
        <fieldset>
                <button
                        className="btn btn-primary"
                        type="submit"
                >
                        Send authentication code
                </button>
        </fieldset>
</Form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ErrorList": [Function],
        "children": Array [
          undefined,
          <fieldset>
            <button
                        className="btn btn-primary"
                        type="submit"
            >
                        Send authentication code
            </button>
</fieldset>,
        ],
        "liveValidate": false,
        "noHtml5Validate": true,
        "noValidate": false,
        "onSubmit": [Function],
        "safeRenderCompletion": false,
        "schema": Object {
          "properties": Object {
            "phoneNumber": Object {
              "title": "What's your phone number",
              "type": "string",
            },
          },
          "required": Array [
            "phoneNumber",
          ],
          "title": "Add SMS Authentication",
          "type": "object",
        },
        "uiSchema": Object {
          "phoneNumber": Object {},
        },
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-primary"
              type="submit"
>
              Send authentication code
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Send authentication code",
              "className": "btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Send authentication code",
            "type": "button",
          },
          "type": "fieldset",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Form
          ErrorList={[Function]}
          liveValidate={false}
          noHtml5Validate={true}
          noValidate={false}
          onSubmit={[Function]}
          safeRenderCompletion={false}
          schema={
                    Object {
                              "properties": Object {
                                "phoneNumber": Object {
                                  "title": "What's your phone number",
                                  "type": "string",
                                },
                              },
                              "required": Array [
                                "phoneNumber",
                              ],
                              "title": "Add SMS Authentication",
                              "type": "object",
                            }
          }
          uiSchema={
                    Object {
                              "phoneNumber": Object {},
                            }
          }
>
          <fieldset>
                    <button
                              className="btn btn-primary"
                              type="submit"
                    >
                              Send authentication code
                    </button>
          </fieldset>
</Form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ErrorList": [Function],
          "children": Array [
            undefined,
            <fieldset>
              <button
                            className="btn btn-primary"
                            type="submit"
              >
                            Send authentication code
              </button>
</fieldset>,
          ],
          "liveValidate": false,
          "noHtml5Validate": true,
          "noValidate": false,
          "onSubmit": [Function],
          "safeRenderCompletion": false,
          "schema": Object {
            "properties": Object {
              "phoneNumber": Object {
                "title": "What's your phone number",
                "type": "string",
              },
            },
            "required": Array [
              "phoneNumber",
            ],
            "title": "Add SMS Authentication",
            "type": "object",
          },
          "uiSchema": Object {
            "phoneNumber": Object {},
          },
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-primary"
                type="submit"
>
                Send authentication code
</button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Send authentication code",
                "className": "btn btn-primary",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Send authentication code",
              "type": "button",
            },
            "type": "fieldset",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
